pkg/sync/sync_context.go

```diff
@@ -502,6 +530,35 @@ func (sc *syncContext) getSyncTasks() (_ syncTasks, successful bool) {
 			task.targetObj = task.targetObj.DeepCopy()
 			task.targetObj.SetNamespace(sc.namespace)
 		}
+		ns := task.targetObj.GetNamespace()
+		if ns != "" {
+			namespaceMap[ns] = ns
+		}
+	}
+
+	if sc.createNamespace && len(namespaceMap) >= 1 {
```

Comment:
Looks like this will auto-create all namespaces referenced in tasks, right?

Namespace auto-creation should auto-create only target sync namespace (specified in `syncContext.namespace` variable).